django/cantusdb_project/main_app/views/chant.py

```diff
@@ -776,56 +798,76 @@ def get_queryset(self) -> QuerySet:
                 # as a substring
                 feasts = Feast.objects.filter(name__icontains=feast)
                 q_obj_filter &= Q(feast__in=feasts)
-            if self.request.GET.get('order'):
-                if self.request.GET.get('order') == 'siglum':
-                    order = 'siglum'
-                elif self.request.GET.get('order') == 'incipit':
+            order: str = 'siglum'
+            order_get_param: Optional[str] = self.request.GET.get('order')
+            sort_get_param: Optional[str] = self.request.GET.get('sort')
+            if order_get_param is not None:
+                if order_get_param == 'siglum':
+                    pass # this is the default
+                elif order_get_param == 'incipit':
                     order = 'incipit'
-                elif self.request.GET.get('order') == 'office':
+                elif order_get_param == 'office':
                     order = 'office'
-                elif self.request.GET.get('order') == 'genre':
+                elif order_get_param == 'genre':
                     order = 'genre'
-                elif self.request.GET.get('order') == 'cantus_id':
+                elif order_get_param == 'cantus_id':
                     order = 'cantus_id'
-                elif self.request.GET.get('order') == 'mode':
+                elif order_get_param == 'mode':
                     order = 'mode'
-                elif self.request.GET.get('order') == 'has_fulltext':
+                elif order_get_param == 'has_fulltext':
                     order = 'manuscript_full_text'
-                elif self.request.GET.get('order') == 'has_melody':
+                elif order_get_param == 'has_melody':
                     order = 'volpiano'
-                elif self.request.GET.get('order') == 'has_image':
+                elif order_get_param == 'has_image':
                     order = 'image_link'
-                else:
-                    order = 'siglum'
-            else:
-                order = 'siglum'
-            if self.request.GET.get('sort'):
-                if self.request.GET.get('sort') == "asc":
-                    order = order
-                elif self.request.GET.get('sort') == 'desc':
-                    order = "-" + order
+
+            if sort_get_param and sort_get_param == "asc":
+                pass # this is the default
```

Comment:
This is a no-op, so you do not need this clause.

Reply:
```py
if sort_get_param and sort_get_param == 'desc':
    order = f"-{order}"
```

This has the exact same effect.

Reply:
Curious, regarding the `if` clause itself: Had I not known from comments on past issues that we're accounting for the case where `sort_get_param` is `None`, I would have been confused by this. I can imagine two other ways to write this:
```python
if sort_get_param == 'desc': # shorter and simpler, since None == 'desc' evaluates to False anyways
```
and
```python
if sort_get_param is not None and sort_get_param == 'desc': # more explicit, since we really care about None-ness rather than truthiness
```
Is there a reason you chose a middle ground here? Is this just convention?

Reply:
Just convention. You rarely find:

```py
if foo is not None:
```

and favour:

```py
if foo:
    # foo is truthy.
```

Unless you're *explicitly* checking against possibly other falsey values:

```py
foo: Optional[bool] = ???  # foo can be None | True | False
if foo is not None:
    # explicit None check makes sure you don't accidentally also capture the case where `foo` is False
```

Reply:
As for your first option, you can do that too -- sure. You just need to watch out in cases like this:

```py
if foo.get("baz") == "bar":
```

since if `foo` is None, you'll crash and burn. But it's not the case here.